### ISSUE TITLE
[Doc] controlled-failover Optimize example code

### ISSUE THREAD

**Issue by zhuhai1221** [apache/pulsar#23512]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### What issue do you find in Pulsar docs?

https://pulsar.apache.org/docs/3.0.x/client-libraries-cluster-level-failover/#controlled-failover
There is unnecessary code present
![image](https://github.com/user-attachments/assets/33ce2f8d-8bfa-44e4-9f97-558a71ea32ff)
![image](https://github.com/user-attachments/assets/3844c552-d3e9-41e1-9ae2-f2bb07886277)

because org.apache.pulsar.client.impl.ClientBuilderImpl#build already initialize
![image](https://github.com/user-attachments/assets/355f274f-c8cb-4d38-a175-b6d4e61a6b2c)

### What is your suggestion?

delete <code>provider.initialize(pulsarClient);<code/>
![image](https://github.com/user-attachments/assets/79976742-1f7c-4070-8106-80b349801bc5)

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by zhuhai1221** [comment]
https://pulsar.apache.org/docs/3.0.x/client-libraries-cluster-level-failover/#automatic-failover
The same problem also exists here
![image](https://github.com/user-attachments/assets/7ec92e1d-4cc7-478f-b2df-719b5d05453d)

**Comment by visxu** [comment]
Yes, I checked the doc and source code, I agree the suggestion. 
Please assign the issue to me.
@crossoverJie

**Comment by Ganny-lp** [comment]
Hello, @david-streamlio , is this issue still relevant? If so, I'd like to work on it.

**Comment by david-streamlio** [comment]
@Ganny-lp This issue is still relevant, and  would appreciate you picking it up.

**Comment by Ganny-lp** [comment]
> [@Ganny-lp](https://github.com/Ganny-lp) This issue is still relevant, and would appreciate you picking it up.

Taking a look at it, it seems that 'provider.initialize' is not present anymore. Therefore, I suggest closing this issue.

<img width="788" height="470" alt="Image" src="https://github.com/user-attachments/assets/24b6ff75-68f9-4455-a889-4bbc5f968b84" />
